### ISSUE TITLE
VMware: HTTP URL based ova deployment is not working using vmware_deploy_ovf module

### ISSUE THREAD

**Issue by Udayendu** [ansible/ansible#61979]
##### SUMMARY
HTTP URL based ova deployment is not working using vmware_deploy_ovf module. Currently vmware_deploy_ovf module is capable of deploying OVA from local server only.

##### ISSUE TYPE
- Provide 'ovf' parameter to get OVA/OVF using HTTP URL.

##### COMPONENT NAME
Module: vmware_deploy_ovf
Parameter: ovf
Currently Supported Method: Path to OVF or OVA file to deploy.

Ex: 
  ovf: '/root/OVA/Demo-1.0.1.ova'

$ pip show ansible
Name: ansible
Version: 2.8.0
Summary: Radically simple IT automation
Home-page: https://ansible.com/
Author: Ansible, Inc.
Author-email: info@ansible.com
License: GPLv3+
Location: /usr/local/lib/python2.7/dist-packages
Requires: jinja2, PyYAML, cryptography

$ pip show pyvmomi
Name: pyvmomi
Version: 6.7.1.2018.12
Summary: VMware vSphere Python SDK
Home-page: https://github.com/vmware/pyvmomi
Author: VMware, Inc.
Author-email: jhu@vmware.com
License: License :: OSI Approved :: Apache Software License
Location: /usr/local/lib/python2.7/dist-packages
Requires: six, requests

##### ADDITIONAL INFORMATION
Keeping the OVA on the same provision server is not possible. Hence pulling it from remote server using HTTP should be a good option to have.

Ex:
  ovf: 'http://ovaserver.example.com/Demo-1.0/Demo-1.0.1.ova'

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/module_utils/vmware`](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/vmware)
* [`lib/ansible/modules/cloud/vmware/vmware_deploy_ovf.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/vmware/vmware_deploy_ovf.py)
* [`lib/ansible/modules/web_infrastructure/deploy_helper.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/web_infrastructure/deploy_helper.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@Udayendu: Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information.

Here are the items we could not find in your description:
- ansible version

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @Akasurde @Tomorrow9 @goneri @lparkes @pgbidkar @ramondelafuente @sivel @warthog9
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@Udayendu, just so you are aware we have a dedicated Working Group for vmware.
You can find other people interested in this in `#ansible-vmware` on Freenode IRC
For more information about communities, meetings and agendas see https://github.com/ansible/community

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Akasurde** [comment]
!component =lib/ansible/modules/cloud/vmware/vmware_deploy_ovf.py

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/cloud/vmware/vmware_deploy_ovf.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/vmware/vmware_deploy_ovf.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sivel** [comment]
Duplicate of https://github.com/ansible/ansible/issues/42666
